zombies.js

```diff
@@ -79,136 +100,224 @@
+    if ((itemToEquip instanceof Weapon) && indexOfWeapon !== -1) {      //if item is a weapon and is in bag
+      if (this.equipped === false) {                                     //if currently unequipped
+        this.equipped = itemToEquip;                                    //item is equipped from bag
+        this.discardItem(itemToEquip);                                  //item is taken out from bag
+      } else {
+        this.getPack().splice(indexOfWeapon, 1, this.equipped);         //swap currently equipped weapon with weapon from bag
+        this.equipped = itemToEquip;
+      }
+    }
+  }
+
+  /**
+   * Player Class Method => eat(itemToEat)
+   * -----------------------------
+   * Player eats a food item, restoring their health.
+   *
+   * Player can only eat Food instances.
+   * Player can only eat food items from their pack.
+   *
+   * Remove itemToEat from the pack.
+   * Increase the player's health by the food's energy amount, but do not
+   *   exceed the player's max health.  If exceeded, simply set player's health
+   *   to max health instead.
+   * To access the player's max health, be sure to use Player's getMaxHealth method.
+   * You should be able to invoke this function on a Player instance.
+   *
+   * @name eat
+   * @param {Food} itemToEat  The food item to eat.
+   */
+
+  eat(itemToEat) {
+    let indexOfFood = this.getPack().indexOf(itemToEat);
+    if ((itemToEat instanceof Food) && indexOfFood !== -1) {
+      if (this.health < this.getMaxHealth()) {
+        this.getPack().splice(itemToEat, 1);
+        this.health += itemToEat.energy;
+        if (this.health > this.getMaxHealth()) {
+          this.health = this.getMaxHealth();
+        }
+      }
+    }
+  }
+
+
+  /**
+   * Player Class Method => useItem(item)
+   * -----------------------------
+   * Player uses an item from the pack.
+   *
+   * If the item is a weapon, the player should equip the item.
+   * If the item is food, the player should eat the item.
+   * You should be able to invoke this function on a Player instance.
+   *
+   * @name useItem
+   * @param {Item/Weapon/Food} item   The item to use.
+   */
+
+  useItem(item) {
+    if (item instanceof Weapon) {
+      this.equip(item);
+    } else {
+      item instanceof Food;
+      this.eat(item);
+    }
+  }
+
+
+  /**
+   * Player Class Method => equippedWith()
+   * -----------------------------
+   * Player checks their equipment.
+   *
+   * Prints the player's name and equipped weapon's name.
+   * If nothing is equipped, prints a message saying so.
+   * Also returns the equipped weapon's name or false if nothing is equipped.
+   * You should be able to invoke this function on a Player instance.
+   *
+   * @name equippedWith
+   * @return {string/boolean}   Weapon name or false if nothing is equipped.
+   */
+
+  equippedWith() {
+    if (typeof (this.equipped) !== 'boolean') {
```

Comment:
This is fine, but a quicker, less wordy check would be to just check `if (this.equipped)` since `this.equipped` can only be `false` or a Weapon.